bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4556,8 +4556,15 @@ ReplicaImp::ReplicaImp(bool firstTime,
 ReplicaImp::~ReplicaImp() {
   // TODO(GG): rewrite this method !!!!!!!! (notice that the order may be important here ).
   // TODO(GG): don't delete objects that are passed as params (TBD)
-  internalThreadPool.stop();
-  postExecThread_.stop();
+  if (!internalThreadPool.isStopped()) {
+    LOG_FATAL(GL, "internalThreadPool should be stopped in ReplicaImp's stop function and not in destructor");
+    ConcordAssert(false);
+  }
+
+  if (!postExecThread_.isStopped()) {
+    LOG_FATAL(GL, "postExecThread_ should be stopped in ReplicaImp's stop function and not in destructor");
+    ConcordAssert(false);
+  }
```

Comment:
Please replace 'delete msg.release();' by 'msg.reset()' in ReplicaImp::messageHandler. I somehow left this code as is in my PR.